lib/api/actions/functions.php

```diff
@@ -91,130 +91,122 @@ function beans_add_smart_action( $hook, $callback, $priority = 10, $args = 1 ) {
 }
 
 /**
- * Modify an action.
+ * Modify one or more of the arguments for the given action, i.e. referenced by its Bean's ID.
  *
- * This function modifies an action registered using {@see beans_add_action()} or
- * {@see beans_add_smart_action()}. Each optional argument must be set to NULL to keep the orginal value.
+ * This function modifies a registered action using {@see beans_add_action()} or
+ * {@see beans_add_smart_action()}. Each optional argument must be set to NULL to keep the original value.
  *
  * The original action can be reset using {@see beans_reset_action()}.
  *
  * @since 1.0.0
- *
- * @param string   $id       The action ID.
- * @param string   $hook     Optional. The name of the new action to which the $callback is hooked.
- *                           Use NULL to keep the original value.
- * @param callback $callback Optional. The name of the new function you wish to be called.
- *                           Use NULL to keep the original value.
- * @param int      $priority Optional. The new priority.
- *                           Use NULL to keep the original value.
- * @param int      $args     Optional. The new number of arguments the function accept.
- *                           Use NULL to keep the original value.
- *
- * @return bool Will always return true.
+ * @since 1.5.0 Made WPCS compliant and improved action parameter filtering.
+ *
+ * @param string        $id       The action's Beans ID, a unique ID tracked within Beans for this action.
+ * @param string|null   $hook     Optional. The new action's event name to which the $callback is hooked.
+ *                                Use NULL to keep the original value.
+ * @param callable|null $callback Optional. The new callback (function or method) you wish to be called.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $priority Optional. The new priority.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $args     Optional. The new number of arguments the $callback accepts.
+ *                                Use NULL to keep the original value.
+ *
+ * @return bool
  */
 function beans_modify_action( $id, $hook = null, $callback = null, $priority = null, $args = null ) {
-
-	// Remove action.
-	if ( $current = _beans_get_current_action( $id ) ) {
-		remove_action( $current['hook'], $current['callback'], $current['priority'], $current['args'] );
+	$action = _beans_build_action_for_valid_args( $hook, $callback, $priority, $args );
+	// If no changes were passed in, there's nothing to modify. Bail out.
+	if ( empty( $action ) ) {
+		return false;
```

Comment:
If no arguments are passed, then there is nothing to modify.  Here, I've moved the entire check to the top of the function.  And if the filtered `$action` is empty, then we bail out.  This move speeds up this function by eliminating unnecessary work:

- getting the current action for the ID.
- removing it via `remove_action`
- storing the empty `$action` in Beans under the "removed" status
- merging the empty `$action` into the current action
- and then adding the action back in via `add_action`